File: PolishPizza/admin/src/service/extraService.ts
import { api as axiosInstance } from '../api/api';
import type { ExtraResponse, ExtraRequest } from '../types/menu';

export const ExtraService = {
    
    getAll: async (): Promise<ExtraResponse[]> => {
        const response = await axiosInstance.get<ExtraResponse[]>('/menu/extras'); 
        return response.data;
    },

    getOne: async (id: string): Promise<ExtraResponse | null> => {
        const response = await axiosInstance.get<ExtraResponse>(`/menu/extras/${id}`);
        return response.data;
    },
    
    update: async (id: string, data: ExtraRequest) => {
        const response = await axiosInstance.put(`/menu/extras/${id}`, data);
        return response.data;
    },
    
    create: async (data: ExtraRequest) => {
        const response = await axiosInstance.post(`/menu/extras`, data);
        return response.data;
    },

    delete: async(id: string): Promise<void> => {
        const response = await axiosInstance.delete(`/menu/extras/${id}`);
        return response.data;
    }
};

File: PolishPizza/menu/src/main/java/pizzeria/menu/extra/service/ExtraService.java
package pizzeria.menu.extra.service;

import pizzeria.menu.extra.dto.response.ExtraResponse;
import pizzeria.menu.extra.dto.request.ExtraPatchRequest;
import pizzeria.menu.extra.dto.request.ExtraRequest;

import java.util.List;
import java.util.UUID;

public interface ExtraService {
    List<ExtraResponse> getAllExtras();

    ExtraResponse getExtraById(UUID id);

    ExtraResponse save(ExtraRequest request);

    void delete(UUID id);

    ExtraResponse update(UUID id, ExtraRequest request);

    ExtraResponse patch(UUID id, ExtraPatchRequest request);
}
